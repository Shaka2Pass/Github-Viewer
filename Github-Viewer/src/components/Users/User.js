import React from 'react';
import PropTypes from 'prop-types';

const User = ({ username }) => (
  <header>
    <div>
      <p>{ username }Username</p>
    </div>
  </header>
);

User.propTypes = {
  username: PropTypes.string.isRequired,
};

export default User;

