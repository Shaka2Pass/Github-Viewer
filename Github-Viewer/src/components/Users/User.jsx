import React from 'react';
import PropTypes from 'prop-types';

const User = ({ name, followers, following, url }) => (
  <div>
    <h3>
      Name: {name}
    </h3>
    <div>
      Followers: {followers}
    </div>
    <div>
      Following: {following}
    </div>
    <div>
      Github Link: {url}
    </div>
  </div>
);

User.propTypes = {
  name: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired, 
};

export default User;

