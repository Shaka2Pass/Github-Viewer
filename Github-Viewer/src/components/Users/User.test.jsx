import React from 'react';
import { shallow } from 'enzyme';
import User from './User';

describe('User', () => {
  it('matches a user', () => {
    const wrapper = shallow(<User username="shaka2pass"/>);
    expect(wrapper).toMatchSnapshot();
  });
});

