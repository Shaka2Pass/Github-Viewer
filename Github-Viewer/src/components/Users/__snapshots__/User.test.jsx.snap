// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches a user 1`] = `
<header>
  <div>
    <p>
      shaka2pass
      Username
    </p>
  </div>
</header>
`;
